src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -277,6 +277,11 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     markRef(current, workInProgress);
 
     if (!shouldUpdate) {
+      // Context providers should defer to sCU for rendering
+      if (hasContext) {
+        invalidateContextProvider(workInProgress, false);
```

Comment:
I guess I misunderstood @bvaughn's original description of this change as a bugfix, or improvement to a more correct behavior. Subsequent comments suggest otherwise.

Now that I think about it, this only affects context providers, right? So I guess functional components aren't affected?